lib/errors/oauth-error.js

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Module dependencies.
+ */
+
+var StandardHttpError = require('standard-http-error');
+var util = require('util');
+
+/**
+ * Constructor.
+ */
+
+function OAuthError(messageOrError, properties) {
+  var message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
+  var error = messageOrError instanceof Error ? messageOrError : null;
+
+  if (error) {
+    properties.inner = error;
```

Comment:
I tend to defend against both - esp in an open source project, it's good to defend against unforseen situations of future modifiers of the code - for example, if someone else writes another method that calls OAuthError but misses the test for it.